src/main/java/ch/admin/bag/covidcertificate/sdk/core/verifier/CertificateVerifier.kt

```diff
@@ -122,63 +124,63 @@ class CertificateVerifier(private val nationalRulesVerifier: NationalRulesVerifi
 	 * @return Outcome state of the revocation check
 	 */
 	private suspend fun checkRevocationStatus(
-		dccHolder: DccHolder,
+		certificateHolder: CertificateHolder,
 		revokedCertificates: RevokedCertificates
 	) = withContext(Dispatchers.Default) {
-		// Revocation is not possible for light certificates, so this check always returns SUCCESS for them
-		if (dccHolder.isLightCertificate()) return@withContext CheckRevocationState.SUCCESS
+		// Revocation is not possible for light certificates, so this check returns the SKIPPED state
+		if (certificateHolder.containsChLightCert()) return@withContext CheckRevocationState.SKIPPED
 
 		try {
 			val revokedCertificateService = RevokedHealthCertService(revokedCertificates)
-			val containsRevokedCertificate = revokedCertificateService.isRevoked(dccHolder.euDGC!!)
+			val containsRevokedCertificate = revokedCertificateService.isRevoked(certificateHolder.certificate as DccCert)
 
 			if (containsRevokedCertificate) {
 				CheckRevocationState.INVALID(ErrorCodes.REVOCATION_REVOKED)
 			} else {
 				CheckRevocationState.SUCCESS
 			}
 		} catch (e: Exception) {
-			CheckRevocationState.ERROR(StateError(ErrorCodes.REVOCATION_UNKNOWN, e.message.toString(), dccHolder))
+			CheckRevocationState.ERROR(StateError(ErrorCodes.REVOCATION_UNKNOWN, e.message.toString(), certificateHolder))
 		}
 	}
 
 	/**
 	 * Checks whether a certificate conforms to a set of national rules
 	 *
-	 * @param dccHolder The object returned from the decoder
+	 * @param certificateHolder The object returned from the decoder
 	 * @param nationalRulesVerifier The verifier responsible to verify a certificate against a national rule set
 	 * @param ruleSet The national rule set from the trust list
 	 * @return Outcome state of the national rules check
 	 */
 	private suspend fun checkNationalRules(
-		dccHolder: DccHolder,
+		certificateHolder: CertificateHolder,
 		nationalRulesVerifier: NationalRulesVerifier,
 		ruleSet: RuleSet
 	) = withContext(Dispatchers.Default) {
 		try {
-			if (dccHolder.isLightCertificate()) {
+			if (certificateHolder.containsChLightCert()) {
 				// If the DccHolder contains a light certificate, the national rules don't have to be verified
 				// In that case, the validity range is from the issuedAt date until the expiration date from the CWT headers
-				val issued = dccHolder.issuedAt?.let { LocalDateTime.ofInstant(it, ZoneId.systemDefault()) }
-				val expiration = dccHolder.expirationTime?.let { LocalDateTime.ofInstant(it, ZoneId.systemDefault()) }
+				val issued = certificateHolder.issuedAt?.let { LocalDateTime.ofInstant(it, ZoneId.systemDefault()) }
+				val expiration = certificateHolder.expirationTime?.let { LocalDateTime.ofInstant(it, ZoneId.systemDefault()) }
 				CheckNationalRulesState.SUCCESS(ValidityRange(issued, expiration))
 			} else {
-				nationalRulesVerifier.verify(dccHolder.euDGC!!, ruleSet)
+				nationalRulesVerifier.verify(certificateHolder.certificate as DccCert, ruleSet)
```

Comment:
In the `CertTypeService` you used an explicit `else if (certificateHolder.containsDccCert()) {`, maybe the same would be clean here as well?